### PR TITLE
Avoid compiler warnings

### PR DESCRIPTION
**Pull request by masterleinad**
While working on #282, I noticed that there are more warnings which this pull request fixes. I wasn't sure if we wanted to enforce this with `-Werror` or similar.

### CODE REVIEW

**Review by stvdwtt — APPROVED** [review]
These look good to me. @Rombur, any concerns?

File: ci/Dockerfile [hunk 1]
```diff
@@ -220,6 +220,7 @@ RUN export TRILINOS_HASH=975307431d60d0859ebaa27c9169cbb1d4287513 && \
         -DTrilinos_ENABLE_SEACAS=ON \
         -DTrilinos_ENABLE_EXPLICIT_INSTANTIATION=ON \
         -DKokkos_ENABLE_CUDA=$CUDA \
+        -DKokkos_ENABLE_CUDA_CONSTEXPR=$CUDA \
```
Comment (Rombur) [inline ci/Dockerfile:223]: Don't modify the Dockerfile. This will unsync the image with the Dockerfile.

**Review by Rombur — CHANGES_REQUESTED** [review]

File: ci/Dockerfile (hunk 1 above)
Comment (masterleinad) [reply ci/Dockerfile:223]: OK. I'll remove but most of the warnings are related to setting this option.

Comment (Rombur) [reply ci/Dockerfile:223]: Then why are you setting this option if it creates a bunch of warnings?

Comment (masterleinad) [reply ci/Dockerfile:223]: It removes all warnings related to calling `constexpr` `host` functions. This is also what we are doing in the `deal.II` CI, see https://github.com/dealii/dealii/blob/3c3910b004cec9be552eb224ceba4bc6fecd47b6/.github/workflows/linux.yml#L335, and we check for this option to be set for an external standalone `Kokkos`, see https://github.com/dealii/dealii/blob/3c3910b004cec9be552eb224ceba4bc6fecd47b6/cmake/modules/FindDEAL_II_KOKKOS.cmake#L53.

**Review by Rombur — APPROVED** [review]

**Pull request merged**